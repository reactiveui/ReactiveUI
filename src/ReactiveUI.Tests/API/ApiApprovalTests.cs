// Copyright (c) 2020 .NET Foundation and Contributors. All rights reserved.
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for full license information.

using System;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using PublicApiGenerator;
using Shouldly;
using Xunit;

namespace ReactiveUI.Tests.API
{
    [ExcludeFromCodeCoverage]
    public class ApiApprovalTests : ApiApprovalBase
    {
        [Fact]
        public void Testing()
        {
            CheckApproval(typeof(Testing.SchedulerExtensions).Assembly);
        }

        [Fact]
        public void ReactiveUI()
        {
            CheckApproval(typeof(RxApp).Assembly);
        }
    }
}
