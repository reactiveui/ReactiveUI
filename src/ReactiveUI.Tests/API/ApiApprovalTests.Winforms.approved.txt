[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.6.1", FrameworkDisplayName=".NET Framework 4.6.1")]
namespace ReactiveUI.Winforms
{
    public class ActivationForViewFetcher : ReactiveUI.IActivationForViewFetcher, Splat.IEnableLogger
    {
        public ActivationForViewFetcher() { }
        public System.IObservable<bool> GetActivationForView(ReactiveUI.IActivatable view) { }
        public int GetAffinityForView(System.Type view) { }
    }
    public class CreatesWinformsCommandBinding : ReactiveUI.ICreatesCommandBinding
    {
        public CreatesWinformsCommandBinding() { }
        public System.IDisposable BindCommandToObject(System.Windows.Input.ICommand command, object target, System.IObservable<object> commandParameter) { }
        public System.IDisposable BindCommandToObject<TEventArgs>(System.Windows.Input.ICommand command, object target, System.IObservable<object> commandParameter, string eventName) { }
        public int GetAffinityForObject(System.Type type, bool hasEventTarget) { }
    }
    public class PlatformOperations
    {
        public PlatformOperations() { }
        public string GetOrientation() { }
    }
    public class ReactiveUserControl<TViewModel> : System.Windows.Forms.UserControl, ReactiveUI.IActivatable, ReactiveUI.IViewFor, ReactiveUI.IViewFor<TViewModel>
        where TViewModel :  class
    {
        public ReactiveUserControl() { }
        [System.ComponentModel.BindableAttribute(true)]
        [System.ComponentModel.CategoryAttribute("ReactiveUI")]
        [System.ComponentModel.DescriptionAttribute("The ViewModel.")]
        [System.ComponentModel.DesignerSerializationVisibilityAttribute(System.ComponentModel.DesignerSerializationVisibility.Hidden)]
        public TViewModel ViewModel { get; set; }
        protected override void Dispose(bool disposing) { }
    }
    public class Registrations
    {
        public Registrations() { }
        public void Register(System.Action<System.Func<object>, System.Type> registerFunction) { }
    }
    [System.ComponentModel.DefaultPropertyAttribute("ViewModel")]
    public class RoutedControlHost : System.Windows.Forms.UserControl, ReactiveUI.INotifyPropertyChanging, ReactiveUI.IReactiveObject, Splat.IEnableLogger, System.ComponentModel.INotifyPropertyChanged
    {
        public RoutedControlHost() { }
        [System.ComponentModel.CategoryAttribute("ReactiveUI")]
        [System.ComponentModel.DescriptionAttribute("The default control when no viewmodel is specified")]
        public System.Windows.Forms.Control DefaultContent { get; set; }
        [System.ComponentModel.CategoryAttribute("ReactiveUI")]
        [System.ComponentModel.DescriptionAttribute("The router.")]
        public ReactiveUI.RoutingState Router { get; set; }
        [System.ComponentModel.BrowsableAttribute(false)]
        public System.IObservable<string> ViewContractObservable { get; set; }
        [System.ComponentModel.BrowsableAttribute(false)]
        public ReactiveUI.IViewLocator ViewLocator { get; set; }
        public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
        public event ReactiveUI.PropertyChangingEventHandler PropertyChanging;
        protected override void Dispose(bool disposing) { }
    }
    [System.ComponentModel.DefaultPropertyAttribute("ViewModel")]
    public class ViewModelControlHost : System.Windows.Forms.UserControl, ReactiveUI.IActivatable, ReactiveUI.INotifyPropertyChanging, ReactiveUI.IReactiveObject, ReactiveUI.IViewFor, Splat.IEnableLogger, System.ComponentModel.INotifyPropertyChanged
    {
        public ViewModelControlHost() { }
        [System.ComponentModel.BindableAttribute(true)]
        [System.ComponentModel.CategoryAttribute("ReactiveUI")]
        [System.ComponentModel.DefaultValueAttribute(true)]
        [System.ComponentModel.DescriptionAttribute("Cache Views")]
        public bool CacheViews { get; set; }
        [System.ComponentModel.BindableAttribute(true)]
        [System.ComponentModel.CategoryAttribute("ReactiveUI")]
        [System.ComponentModel.DescriptionAttribute("The Current View")]
        public object Content { get; set; }
        public System.Windows.Forms.Control CurrentView { get; }
        public static bool DefaultCacheViewsEnabled { get; set; }
        [System.ComponentModel.CategoryAttribute("ReactiveUI")]
        [System.ComponentModel.DescriptionAttribute("The default control when no viewmodel is specified")]
        public System.Windows.Forms.Control DefaultContent { get; set; }
        [System.ComponentModel.BrowsableAttribute(false)]
        public System.IObservable<string> ViewContractObservable { get; set; }
        [System.ComponentModel.BrowsableAttribute(false)]
        public ReactiveUI.IViewLocator ViewLocator { get; set; }
        [System.ComponentModel.BindableAttribute(true)]
        [System.ComponentModel.CategoryAttribute("ReactiveUI")]
        [System.ComponentModel.DescriptionAttribute("The viewmodel to host.")]
        public object ViewModel { get; set; }
        public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
        public event ReactiveUI.PropertyChangingEventHandler PropertyChanging;
        protected override void Dispose(bool disposing) { }
    }
    public class WinformsCreatesObservableForProperty : ReactiveUI.ICreatesObservableForProperty, Splat.IEnableLogger
    {
        public WinformsCreatesObservableForProperty() { }
        public int GetAffinityForObject(System.Type type, string propertyName, bool beforeChanged = False) { }
        public System.IObservable<ReactiveUI.IObservedChange<object, object>> GetNotificationForProperty(object sender, System.Linq.Expressions.Expression expression, string propertyName, bool beforeChanged = False) { }
    }
}